crates/eframe/src/web/events.rs

```diff
@@ -422,8 +421,17 @@ fn install_mousedown(runner_ref: &WebRunner, target: &EventTarget) -> Result<(),
 }
 
 /// Returns true if the cursor is above the canvas, or if we're dragging something.
-fn is_interested_in_pointer_event(egui_ctx: &egui::Context, pos: egui::Pos2) -> bool {
-    egui_ctx.input(|i| i.screen_rect().contains(pos) || i.pointer.any_down() || i.any_touches())
+/// Pass in the position in browser viewport coordinates (usually event.clientX/Y).
+fn is_interested_in_pointer_event(runner: &AppRunner, pos: egui::Pos2) -> bool {
+    window()
+        .unwrap()
+        .document()
+        .unwrap()
+        .element_from_point(pos.x, pos.y)
+        .is_some_and(|element| element.eq(runner.canvas()))
+        || runner
+            .egui_ctx()
+            .input(|i| i.pointer.any_down() || i.any_touches())
```

Comment:
This is very difficult to parse. Naming intermediates helps

```suggestion
    let document = web_sys::window().unwrap().document().unwrap();
    let hovering_canvas = document
        .element_from_point(pos.x, pos.y)
        .is_some_and(|element| element.eq(runner.canvas()));
    let pointer_down = runner
        .egui_ctx()
        .input(|i| i.pointer.any_down() || i.any_touches())
    
    is_hovering_canvas || pointer_down
```

Reply:
Much better!